server/odc-service/src/main/java/com/oceanbase/odc/service/structurecompare/StructureComparisonTask.java

```diff
@@ -0,0 +1,166 @@
+    private final String STRUCTURE_COMPARISON_RESULT_FILE_NAME = "structure-comparison-result.sql";
+    private final String STRUCTURE_COMPARISON_BUCKET_NAME = "structure-comparison";
+
+    public StructureComparisonTask(@NonNull DBStructureComparisonConfig srcConfig,
+            @NonNull DBStructureComparisonConfig tgtConfig,
+            @NonNull Long taskId,
+            @NonNull Long structureComparisonTaskId,
+            @NonNull User user,
+            @NonNull StructureComparisonTaskRepository structureComparisonTaskRepository,
+            @NonNull StructureComparisonTaskResultRepository structureComparisonTaskResultRepository,
+            @NonNull ObjectStorageFacade objectStorageFacade) {
+        this.srcConfig = srcConfig;
+        this.tgtConfig = tgtConfig;
+        initTaskResult(structureComparisonTaskId);
+        this.taskId = taskId;
+        this.user = user;
+        this.structureComparisonTaskResultRepository = structureComparisonTaskResultRepository;
+        this.structureComparisonTaskRepository = structureComparisonTaskRepository;
+        this.objectStorageFacade = objectStorageFacade;
+    }
+
+    private void initTaskResult(Long structureComparisonTaskId) {
+        this.taskResult.setTaskId(structureComparisonTaskId);
+        if (!srcConfig.getBlackListMap().isEmpty()) {
+            this.taskResult.setComparingList(srcConfig.getBlackListMap().get(DBObjectType.TABLE).stream()
+                    .map(tableName -> new Comparing(DBObjectType.TABLE, tableName))
+                    .collect(Collectors.toList()));
+        }
+    }
+
+    @Override
+    public DBStructureComparisonTaskResult call() throws Exception {
+        SecurityContextUtils.setCurrentUser(user);
+        StructureComparisonTraceContextHolder.trace(user.getId(), taskId);
+        try {
+            log.info("Structure comparison task starts, id={}", taskResult.getTaskId());
+            long startTimestamp = System.currentTimeMillis();
+            taskResult.setStatus(TaskStatus.RUNNING);
+            try {
+                List<DBObjectComparisonResult> results = comparator.compare(srcConfig, tgtConfig);
+                saveComparisonResult(results, tgtConfig.getConnectType().getDialectType());
+                taskResult.setStatus(TaskStatus.DONE);
+            } catch (Exception e) {
+                log.warn("Failed to run structure comparison task, id={}, error message={}", taskResult.getTaskId(),
+                        e.getMessage(), e);
+                taskResult.setStatus(TaskStatus.FAILED);
+            }
+            log.info("Structure comparison task ends, id={}, task status={}, time consuming={} seconds",
+                    taskResult.getTaskId(), taskResult.getStatus(),
+                    (System.currentTimeMillis() - startTimestamp) / 1000);
+            return taskResult;
+        } finally {
+            try {
+                ((AutoCloseable) srcConfig.getDataSource()).close();
```

Comment:
please check if this datasource is instance of a `AutoCloseable` before you attempt to close it